utils/parse.js

```diff
@@ -7,34 +7,27 @@ const fs = require('fs');
 
 const log = require('debug')('eslint-plugin-import:parse');
 
-function getBabelVisitorKeys(parserPath) {
+function getBabelEslintVisitorKeys(parserPath) {
   if (parserPath.endsWith('index.js')) {
     const hypotheticalLocation = parserPath.replace('index.js', 'visitor-keys.js');
     if (fs.existsSync(hypotheticalLocation)) {
       const keys = moduleRequire(hypotheticalLocation);
       return keys.default || keys;
     }
-  } else if (parserPath.endsWith('index.cjs')) {
```

Comment:
was this branch always only for the new Babel parser?

Reply:
Yes, introduced by https://github.com/babel/babel/pull/13199

Reply:
I’m confused; that seems to be for an unreleased Babel version. This branch here, in this plugin, is presumably for a released Babel version?

Reply:
Specifically, this was added in #1660 / #2212 by @Hypnosphi - any idea why it was needed then, but not needed after this PR?

Reply:
Apparently, it targeted Babel 7.14: https://github.com/Hypnosphi/eslint-plugin-import/commit/513960d9f559fc98bc15a8215e4a63814e6f63d7

Reply:
@ljharb Currently we have both the versions in `@babel/eslint-parser`: the "default one" which works with this PR (because the internal file can be directly required), and the "worker one" which wouldn't work (it can be used with `@babel/eslint-parser/experimental-worker`).

The worker one will be enabled by default in Babel 8 (not released yet) and it will be the only one available. So this PR is mostly to future-proof this implementation, and to use the public `@babel/eslint-parser` API rather than its internal files (where we don't make any stability guarantee).

Reply:
ok - so for Babel 7, the PR is still supporting both parsers because they also provide visitorKeys, and in Babel 8, it will still work for the same reason - but the current way supports both parsers in Babel 7, but will break in Babel 8 due to “exports”?

Reply:
This PR works with:
- `babel-eslint` (which is no more maintained, so it will never have breaking changes and it will continue working forever)
- With `@babel/eslint-parser` 7 and 8, because it uses the `visitorKeys` option

I didn't actually think about `"exports"`: the breakage I found with the current approach in Babel 8 is that you cannot just require `@babel/eslint-parser/lib/worker/visitor-keys.cjs`, but you have to first `await require("@babel/eslint-parser/lib/worker/babel-core.js").init")` and _then_ you can require `@babel/eslint-parser/lib/worker/visitor-keys.cjs`.
We will also introduce `"exports"` in Babel 8, but since `eslint-plugin-import` uses absolute paths I don't think that it will be affected.

Reply:
All of our absolute paths come from `require.resolve`, so anything not in "exports" wouldn't resolve in the first place.